packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -149,17 +150,27 @@ export class WorkspaceFrontendContribution implements CommandContribution, Keybi
             isEnabled: () => this.workspaceService.isMultiRootWorkspaceEnabled,
             execute: () => this.saveWorkspaceAs()
         });
-        commands.registerCommand(WorkspaceCommands.SAVE_AS,
-            UriAwareCommandHandler.MonoSelect(this.selectionService, {
-                execute: (uri: URI) => this.saveAs(uri),
-            }));
+        commands.registerCommand(WorkspaceCommands.SAVE_AS, {
+            isEnabled: () => this.canBeSavedAs(this.applicationShell.currentWidget),
+            execute: () => {
```

Comment:
I just wanted to mention that with this approach we lose the following abilities:

- we lose the ability to rely on the selection in order to perform `save_as` and now rely on the widget
- we lose the ability to callers to execute the command with a passed uri like so:


```ts
registry.registerCommand({ id: 'test-saveAs', label: 'test-saveAs' }, {
    execute: async () => {
        await registry.executeCommand(WorkspaceCommands.SAVE_AS.id, new URI('/tmp/a.ts'));
    }
});
```

If we are fine with this behavior, we should probably document it as such in the changelog's breaking changes.

Reply:
Despite the old change including the `UriAware...` handler, since it used a URI from one place and the `currentWidget` at the same time it lead to weird cases (e.g. passed `uri` to the command doesn't match the `currentWidget`'s uri).

So either we definitely break that behavior, or this PR requires a bit more changes to address this use case?

Reply:
@marechal-p I'd be fine to break the behavior and align more closely to [vscode's implementation](https://github.com/microsoft/vscode/blob/9c78fa40ca7ab6b58d0ab5cd33858a39a2b7102d/src/vs/workbench/contrib/files/browser/fileCommands.ts#L371-L425) which also relies on the `currentEditor`. I wanted to bring it up so that we properly document the breaking change and the fact we cannot call the method anymore the way I wrote in the snippet.

Reply:
I don't see a nice way to handle saving a random URI and properly handling eventual opened editors. @Darkgaja wdyt?

Reply:
I think so too. This was the reason I implemented it for the `currentWidget` in the first place.

Reply:
@marechal-p @vince-fugnitto I rebased my branch and documented the breaking change.